src/codes.ts

```diff
@@ -43,6 +45,15 @@ const codeList: ReadonlyArray<Readonly<ICode>> = Object.freeze((
 
 export const ChainType = tcomb.irreducible('ChainType', (n) => isValid(n));
 
+const chainCodeFromUnspentType = (u: UnspentType, p = Purpose.external): ChainCode | undefined => {
```

Comment:
I usually use `from` when creating a new instance of something and `for` when we have a mapping.

How about `Codes.forType(t): CodesByPurpose` or `Codes.external.forType(t): ChainCode`? 

We don't default to internal or external anywhere else and it would be nice if we could keep it that way. Also we can shorten `UnspentType` to just `Type` because we already have the `CodesByType` classes.

Reply:
Done.